### PR TITLE
Remove child from parent context correctly in destroy_context

### PR DESCRIPTION
**Pull request by sree-revoori1**
Previously, we were always setting the children of a parent context to 0 in destroy_context, which is incorrect: If P is the parent of A and B and we destroy_context on A, then B would cease to be the child of P.

### CODE REVIEW

File: dpe/src/commands/destroy_context.rs [hunk 1]
```diff
@@ -57,8 +57,8 @@ impl CommandExecution for DestroyCtxCmd {
             .take(contexts_to_destroy.len())
         {
             if *context_to_destroy {
-                dpe.contexts[idx].destroy();
                 let parent = dpe.contexts[idx].parent_idx as usize;
+                dpe.contexts[idx].destroy();
```
Comment (jhand2) [inline dpe/src/commands/destroy_context.rs:61]: nit: in Context::destroy should we set parent_idx to 0xFF to protect against this code being moved around incorrectly in the future?

File: dpe/src/commands/destroy_context.rs [hunk 2]
```diff
@@ -33,7 +33,7 @@ impl CommandExecution for DestroyCtxCmd {
         // contexts_to_destroy[i] == true implies dpe.contexts[i] must be destroyed
         let mut contexts_to_destroy = [false; MAX_HANDLES];
         // updated_children[i] holds i's non-destroyed children.
-        let mut updated_children = [0u32; MAX_HANDLES];
+        let mut updated_children = [u32::MAX; MAX_HANDLES];
```
Comment (jhand2) [inline dpe/src/commands/destroy_context.rs:36]: I think this code works correctly, but is a bit hard to follow, which could lead to issues being introduced in the future. I'll think about this a little more, but maybe we could try to refactor this to simplify it.

File: dpe/src/commands/destroy_context.rs (hunk 1 above)
Comment (jhand2) [inline dpe/src/commands/destroy_context.rs:61]: Instead of `dpe.contexts[idx]` can we use `context_to_destroy`?

File: dpe/src/commands/destroy_context.rs (hunk 2 above)
Comment (jhand2) [reply dpe/src/commands/destroy_context.rs:36]: Here's a straw proposal for how this could maybe be changed:

```rust
// TODO: Is there a way to be extra sure this is being copied and not moved?
// Like calling a .copy function?
let mut tmp_contexts = dpe.contexts;

let to_destroy = (1 << idx) | dpe.get_descendants(context)?;

for idx, c in tmp_contexts {
    // Clears all the to_destroy bits in the children of every context
    c.children &= ~to_destroy;
    if to_destroy & (1 << idx) != 0 {
        c.destroy();
    }
}

dpe.contexts = tmp_contexts;
```

I _think_ this does the same thing as the existing code.

Looking at this, I'm not even sure we need to do the copying because I don't think any of these calls can fail.

File: dpe/src/commands/destroy_context.rs (hunk 1 above)
Comment (sree-revoori1) [reply dpe/src/commands/destroy_context.rs:61]: Done

Comment (sree-revoori1) [reply dpe/src/commands/destroy_context.rs:61]: context_to_destroy is just a bool, so how would we use it?

Comment (jhand2) [reply dpe/src/commands/destroy_context.rs:61]: Oh, never mind, I misread and thought it was the actual context.

File: dpe/src/commands/destroy_context.rs (hunk 2 above)
Comment (sree-revoori1) [reply dpe/src/commands/destroy_context.rs:36]: Thanks, this is way simpler. Also removed the copying.

**Review by jhand2 — APPROVED** [review]
